hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/ozoneimpl/OnDemandContainerScanner.java

```diff
@@ -135,13 +137,18 @@ private static void performOnDemandScan(Container<?> container) {
       ContainerData containerData = container.getContainerData();
       logScanStart(containerData);
 
-      ScanResult result =
-          container.scanData(instance.throttler, instance.canceler);
-      // Metrics for skipped containers should not be updated.
-      if (result.getFailureType() == DELETED_CONTAINER) {
-        LOG.error("Container [{}] has been deleted.",
-            containerId, result.getException());
-        return;
+      ScanResult result = ScanResult.healthy();
+      // OPEN containers are scanned here for metadata only
+      if (container.shouldScanMetadata() && !container.shouldScanData()) {
+        result = container.scanMetaData();
```

Comment:
Not necessary, but if you want, you may add a debug log.
`LOG.debug("Performing metadata-only scan for open container {}", container.getContainerID());`